assets/js/blocks/classic-template/constants.ts

```diff
@@ -67,6 +67,14 @@ export const TEMPLATES: TemplateDetails = {
 		),
 		placeholder: PLACEHOLDERS.archiveProduct,
 	},
+	'taxonomy-product_': {
+		type: TYPES.productTaxonomy,
+		title: __(
+			"WooCommerce Product's Custom Taxonomy Block",
+			'woo-gutenberg-products-block'
+		),
+		placeholder: PLACEHOLDERS.archiveProduct,
+	},
```

Comment:
Note that this needs to be the last "taxonomy-product_" item after all the other "taxonomy-product_" items. If this is the first item, then it would cause unwanted behavior because of the partial matching in the for-loop in `getTemplateDetailsBySlug` function.